src/docs/slack-tracking.md

```diff
@@ -0,0 +1,399 @@
+number of in-object properties used by any descendents in the tree of maps below the root map.
+
+Below shows the map family after running the code above, and of course, slack tracking is complete:
+
+<figure>
+  <img src="/_img/docs/slack-tracking/root-map-3.svg" loading="lazy">
+  <figcaption>
+  <i>(X, Y, Z) means (instance size, number of in-object properties, number of unused properties)</i>
+  </figcaption>
+</figure>
+
+## Optimized code
+
+How about if we compile optimized code before slack tracking is finished? We'll use a handy native syntax command `%OptimizeFunctionOnNextCall()` to force a optimized compile to go forward before we finished slack tracking (*be sure to add the flag `--no-lazy-feedback-allocation` when using this function, to make sure that your function gets a feedback vector right away*):
+
+```javascript
+function foo(a1, a2, a3, a4) {
+  return new Peak(a1, a2, a3, a4);
+}
+
+let m1 = foo("Wendelstein", 1838);
+let m2 = foo("Matterhorn", 4478, 1040, true);
+%OptimizeFunctionOnNextCall(foo);
+foo("Zugspitze", 2962);
+```
+
+That should be enough to compile and run optimized code. We do something in TurboFan (the optimizing compiler) called [**Create Lowering**](https://source.chromium.org/chromium/chromium/src/+/master:v8/src/compiler/js-create-lowering.h;l=32;drc=ee9e7e404e5a3f75a3ca0489aaf80490f625ca27), where we inline the allocation of objects. That means the native code we produce will emit instructions to ask the GC for the instance size of the object to allocate, and then carefully initialize those fields. But this code would be invalid if slack tracking were to stop at some later point.
+
+To avoid this problem what we do is just say sorry, we are ending slack tracking early. This makes sense because normally, we wouldn't compile an optimized function until thousands of objects have been created. So slack tracking *should* be finished. If it's not, too bad! The object must not be that important anyway if less than 7 of them have been created by this point. Normally, remember, we are only optimizing after the program ran for a long time.
+
+### But wait, it's not that easy
+
+We can compile optimized code on the main thread, in which case we can get away with prematurely ending slack tracking with some calls to change the root map. However, we do as much compilation as possible on a background thread. From this thread it would be dangerous to touch the initial map because it might be changing on the main thread. So our technique is like this:
+
+ 1. **Guess** that the instance size will be what it would be if you did stop slack tracking right now. Remember this size.
+ 2. When the compilation is almost done, we return to the main thread where we can safely force completion of slack tracking if it wasn't already done.
+ 3. Check: is the instance size what we predicted? If so, **we are good!** If not, throw away the code object and head to the bar to drown our sorrows.
```

Comment:
To keep it informative, you can write something like:
"~~head to the bar to drown our sorrows~~ reoptimize the function"

Reply:
Thanks, I changed it to "try again later."